proto-grpc.md

```diff
@@ -0,0 +1,254 @@
+# Protobuf and gRPC with Bazel
+
+The use of [Protobuf](https://developers.google.com/protocol-buffers) and
+[gRPC](https://grpc.io) with [Bazel](https://bazel.build) is a particularly common scenario, since
+these three technologies provide a complimentary base of build, serialisation and remote-call upon
+which to build complex systems and products. This document summarises the current state of the
+Protobuf/gRPC Bazel ecosystem and gives advice on getting started when you are looking to integrate
+one of these tools into your project.
+
+Given that all three of Bazel, gRPC and Protobuf are Google-derived projects, you may expect that
+there is a simple integration between them. However, the existence of this document should indicate
+that there is not a single simple answer to integrating these tools. Originally, Protobuf and gRPC
+rules were baked into Bazel for a few key languages, but this has understandably not continued to
+scale with the many new langauges that are now supported in external Bazel rulesets. As a result,
+independent rulesets have been released by the community that provide the integrations necessary to
+convert `.proto` files into a wide variety of languages and libraries.
+
+This page is expected to continue as a 'live' document; as the ecosystem evolves and new rulesets
+are released, this document can be edited at
+[SIG-rules-authors](https://github.com/bazel-contrib/SIG-rules-authors/tree/gh-pages) to continue to
+reflect today's state.
+
+
+## The Basics
+
+At its most basic level, the task of working with Protobuf and gRPC `.proto` files is much the same
+as elsewhere: we execute the Protobuf Compiler `protoc` with the necessary arguments, environment
+and plugins to get it to produce a desired set of files or libraries for any specific language. For
+those unfamiliar with protoc, it is a small compiler executable that parses sets of `.proto`
+formatted files and executes plugins to write a set of output source code files. These files
+implement the interfaces for working with the messages and services in the language supported by the
+plugin.
+
+However, given the prevalence of Protobuf and gRPC use, users are not expected to reinvent the rules
+needed for calling protoc themselves. Instead, the common base dependency ruleset
+[bazelbuild/rules_proto](https://github.com/bazelbuild/rules_proto) provides the `proto_library`
+rule which can be used to establish libraries of `.proto` files and map their dependencies. For
+example, given the following dependency tree of `.proto` files, we could define a set of
+`proto_library` targets:
+
+![Tree of proto files](./assets/proto-tree.png)
```

Comment:
nit: maybe it would be nice to give a `bazel query --output=dot` one-liner so ppl know how to reproduce such a diagram

Reply:
Added